### ISSUE TITLE
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c:363:12: error: 'i2c_slave_struct_t' {aka 'struct i2c_slave_struct_t'} has no member named 'lock'

### ISSUE THREAD

**Issue by rhr407** [espressif/arduino-esp32#5943]
I am facing the following error after adding arduino as an esp-idf component. The version of esp-idf is v4.4
```
FAILED: esp-idf/arduino/CMakeFiles/__idf_arduino.dir/cores/esp32/esp32-hal-i2c-slave.c.obj 
/home/rizwan/.espressif/tools/xtensa-esp32-elf/esp-2021r2-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc -DHAVE_CONFIG_H -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -Iconfig -I../components/arduino/variants/esp32 -I../components/arduino/cores/esp32 -I../components/arduino/libraries/ArduinoOTA/src -I../components/arduino/libraries/AsyncUDP/src -I../components/arduino/libraries/BLE/src -I../components/arduino/libraries/BluetoothSerial/src -I../components/arduino/libraries/DNSServer/src -I../components/arduino/libraries/EEPROM/src -I../components/arduino/libraries/ESP32/src -I../components/arduino/libraries/ESPmDNS/src -I../components/arduino/libraries/FFat/src -I../components/arduino/libraries/FS/src -I../components/arduino/libraries/HTTPClient/src -I../components/arduino/libraries/HTTPUpdate/src -I../components/arduino/libraries/LittleFS/src -I../components/arduino/libraries/NetBIOS/src -I../components/arduino/libraries/Preferences/src -I../components/arduino/libraries/RainMaker/src -I../components/arduino/libraries/SD_MMC/src -I../components/arduino/libraries/SD/src -I../components/arduino/libraries/SimpleBLE/src -I../components/arduino/libraries/SPIFFS/src -I../components/arduino/libraries/SPI/src -I../components/arduino/libraries/Ticker/src -I../components/arduino/libraries/Update/src -I../components/arduino/libraries/USB/src -I../components/arduino/libraries/WebServer/src -I../components/arduino/libraries/WiFiClientSecure/src -I../components/arduino/libraries/WiFi/src -I../components/arduino/libraries/WiFiProv/src -I../components/arduino/libraries/Wire/src -I../components/arduino/cores/esp32/libb64 -I/home/rizwan/esp/esp-idf/components/newlib/platform_include -I/home/rizwan/esp/esp-idf/components/freertos/include -I/home/rizwan/esp/esp-idf/components/freertos/include/esp_additions/freertos -I/home/rizwan/esp/esp-idf/components/freertos/port/xtensa/include -I/home/rizwan/esp/esp-idf/components/freertos/include/esp_additions -I/home/rizwan/esp/esp-idf/components/esp_hw_support/include -I/home/rizwan/esp/esp-idf/components/esp_hw_support/include/soc -I/home/rizwan/esp/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/rizwan/esp/esp-idf/components/esp_hw_support/port/esp32/. -I/home/rizwan/esp/esp-idf/components/heap/include -I/home/rizwan/esp/esp-idf/components/log/include -I/home/rizwan/esp/esp-idf/components/lwip/include/apps -I/home/rizwan/esp/esp-idf/components/lwip/include/apps/sntp -I/home/rizwan/esp/esp-idf/components/lwip/lwip/src/include -I/home/rizwan/esp/esp-idf/components/lwip/port/esp32/include -I/home/rizwan/esp/esp-idf/components/lwip/port/esp32/include/arch -I/home/rizwan/esp/esp-idf/components/soc/include -I/home/rizwan/esp/esp-idf/components/soc/esp32/. -I/home/rizwan/esp/esp-idf/components/soc/esp32/include -I/home/rizwan/esp/esp-idf/components/hal/esp32/include -I/home/rizwan/esp/esp-idf/components/hal/include -I/home/rizwan/esp/esp-idf/components/hal/platform_port/include -I/home/rizwan/esp/esp-idf/components/esp_rom/include -I/home/rizwan/esp/esp-idf/components/esp_rom/include/esp32 -I/home/rizwan/esp/esp-idf/components/esp_rom/esp32 -I/home/rizwan/esp/esp-idf/components/esp_common/include -I/home/rizwan/esp/esp-idf/components/esp_system/include -I/home/rizwan/esp/esp-idf/components/esp_system/port/soc -I/home/rizwan/esp/esp-idf/components/esp_system/port/public_compat -I/home/rizwan/esp/esp-idf/components/esp32/include -I/home/rizwan/esp/esp-idf/components/xtensa/include -I/home/rizwan/esp/esp-idf/components/xtensa/esp32/include -I/home/rizwan/esp/esp-idf/components/driver/include -I/home/rizwan/esp/esp-idf/components/driver/esp32/include -I/home/rizwan/esp/esp-idf/components/esp_pm/include -I/home/rizwan/esp/esp-idf/components/esp_ringbuf/include -I/home/rizwan/esp/esp-idf/components/efuse/include -I/home/rizwan/esp/esp-idf/components/efuse/esp32/include -I/home/rizwan/esp/esp-idf/components/vfs/include -I/home/rizwan/esp/esp-idf/components/esp_wifi/include -I/home/rizwan/esp/esp-idf/components/esp_event/include -I/home/rizwan/esp/esp-idf/components/esp_netif/include -I/home/rizwan/esp/esp-idf/components/esp_eth/include -I/home/rizwan/esp/esp-idf/components/tcpip_adapter/include -I/home/rizwan/esp/esp-idf/components/esp_phy/include -I/home/rizwan/esp/esp-idf/components/esp_phy/esp32/include -I/home/rizwan/esp/esp-idf/components/esp_ipc/include -I/home/rizwan/esp/esp-idf/components/app_trace/include -I/home/rizwan/esp/esp-idf/components/esp_timer/include -I/home/rizwan/esp/esp-idf/components/spi_flash/include -I/home/rizwan/esp/esp-idf/components/mbedtls/port/include -I/home/rizwan/esp/esp-idf/components/mbedtls/mbedtls/include -I/home/rizwan/esp/esp-idf/components/mbedtls/esp_crt_bundle/include -I/home/rizwan/esp/esp-idf/components/mdns/include -I/home/rizwan/esp/esp-idf/components/console -I/home/rizwan/esp/esp-idf/components/esp_adc_cal/include -I/home/rizwan/esp/esp-idf/components/wifi_provisioning/include -I/home/rizwan/esp/esp-idf/components/protocomm/include/common -I/home/rizwan/esp/esp-idf/components/protocomm/include/security -I/home/rizwan/esp/esp-idf/components/protocomm/include/transports -I/home/rizwan/esp/esp-idf/components/bt/common/osi/include -I/home/rizwan/esp/esp-idf/components/bt/include/esp32/include -I/home/rizwan/esp/esp-idf/components/bt/common/api/include/api -I/home/rizwan/esp/esp-idf/components/bt/common/btc/profile/esp/blufi/include -I/home/rizwan/esp/esp-idf/components/bt/common/btc/profile/esp/include -I/home/rizwan/esp/esp-idf/components/bt/host/bluedroid/api/include/api -I/home/rizwan/esp/esp-idf/components/nvs_flash/include -I/home/rizwan/esp/esp-idf/components/nghttp/port/include -I/home/rizwan/esp/esp-idf/components/nghttp/nghttp2/lib/includes -I/home/rizwan/esp/esp-idf/components/fatfs/diskio -I/home/rizwan/esp/esp-idf/components/fatfs/vfs -I/home/rizwan/esp/esp-idf/components/fatfs/src -I/home/rizwan/esp/esp-idf/components/wear_levelling/include -I/home/rizwan/esp/esp-idf/components/sdmmc/include -I/home/rizwan/esp/esp-idf/components/app_update/include -I/home/rizwan/esp/esp-idf/components/bootloader_support/include -I/home/rizwan/esp/esp-idf/components/spiffs/include -I/home/rizwan/esp/esp-idf/components/openssl/include -I/home/rizwan/esp/esp-idf/components/esp_hid/include -mlongcalls -Wno-frame-address   -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -ggdb -Og -fmacro-prefix-map=/mnt/sda2/Insync/rhr407@gmail.com/Google-Drive/SEMPHOS/Projects/OnGoing/ValarSystems/Rizwan/Project_v5/model-h-idf-fastaccelstepper=. -fmacro-prefix-map=/home/rizwan/esp/esp-idf=IDF -fstrict-volatile-bitfields -Wno-error=unused-but-set-variable -fno-jump-tables -fno-tree-switch-conversion -std=gnu99 -Wno-old-style-declaration -D_GNU_SOURCE -DIDF_VER=\"v4.4-dev-3703-gddc44956bf\" -DESP_PLATFORM -D_POSIX_READER_WRITER_LOCKS -DARDUINO=10812 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD=\"ESP32_DEV\" -DARDUINO_VARIANT=\"esp32\" -DESP32 -MD -MT esp-idf/arduino/CMakeFiles/__idf_arduino.dir/cores/esp32/esp32-hal-i2c-slave.c.obj -MF esp-idf/arduino/CMakeFiles/__idf_arduino.dir/cores/esp32/esp32-hal-i2c-slave.c.obj.d -o esp-idf/arduino/CMakeFiles/__idf_arduino.dir/cores/esp32/esp32-hal-i2c-slave.c.obj   -c ../components/arduino/cores/esp32/esp32-hal-i2c-slave.c
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c: In function 'i2cSlaveDeinit':
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c:363:12: error: 'i2c_slave_struct_t' {aka 'struct i2c_slave_struct_t'} has no member named 'lock'
     if(!i2c->lock){
            ^~
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c: In function 'i2cSlaveWrite':
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c:380:12: error: 'i2c_slave_struct_t' {aka 'struct i2c_slave_struct_t'} has no member named 'lock'
     if(!i2c->lock){
            ^~
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c: In function 'i2c_slave_isr_handler':
../components/arduino/cores/esp32/esp32-hal-i2c-slave.c:694:13: warning: unused variable 'tx_fifo_len' [-Wunused-variable]
     uint8_t tx_fifo_len = SOC_I2C_FIFO_LEN - i2c_ll_get_txfifo_len(i2c->dev);
             ^~~~~~~~~~~
[1281/1502] Building C object esp-idf/libsodium/CMakeFiles/__idf_libsodium.dir/libsodium/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1

```

**Comment by me-no-dev** [comment]
yup :) will fix. you can enable hal locks and should compile fine.

**Comment by rhr407** [comment]
Thanks, it's working now.

**Comment by VojtechBartoska** [comment]
probably will be solved by #6108

**Comment by VojtechBartoska** [comment]
@rhr407 Are you able to test your issue on development version [2.0.3-RC1](https://github.com/espressif/arduino-esp32/releases/tag/2.0.3-RC1) to check if this is still valid? You can take a look on [Docs](https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html) where is explained how to choose development release version in Arduino IDE.
